.semaphore/semaphore.yml

```diff
@@ -19,60 +19,164 @@ queue:
+      epilogue: *build-test-epilogue
+
+  # Disabled until we can connect to an active GUI session
+  # - name: "Build & Test (VS Code: darwin x64)"
+  #   dependencies: []
+  #   skip: *build-test-skip-stable
+  #   task:
+  #     agent:
+  #       machine:
+  #         type: s1-prod-macos-13-5-amd64
+  #     prologue: *build-test-prologue
+  #     jobs: *build-test-jobs-stable
+  #     epilogue: *build-test-epilogue
+
+  # - name: "Build & Test (VS Code: darwin arm64)"
+  #   dependencies: []
+  #   skip: *build-test-skip-stable
+  #   task:
+  #     agent:
+  #       machine:
+  #         type: s1-prod-macos-13-5-arm64
+  #     prologue: *build-test-prologue
+  #     jobs: *build-test-jobs-stable
+  #     epilogue: *build-test-epilogue
+
+  - name: "Build & Test (VS Code: win32 x64)"
+    dependencies: []
+    skip: *build-test-skip-stable
     task:
+      agent:
+        machine:
+          type: s1-prod-windows
+      prologue:
+        commands:
+          - checkout
+          - make ci-bin-sem-cache-restore
+          - $Env:PATH += ";C:\Program Files\Git\bin"
+          - $Env:PATH += ";C:\Program Files (x86)\Windows Kits\10\bin\10.0.19041.0\x64"
+          # Ensure npm and npx are available
+          - npm --version
+          - npx --version
+          - choco install gh -y
+          - $Env:PATH += ";C:\Program Files\GitHub CLI\"
+          - gh --version
+          - |
+            if (Test-Path "$HOME/.gh-token") {
+              $Env:GH_TOKEN = Get-Content "$HOME/.gh-token" -Raw | ForEach-Object { $_.Trim() }
+              if ([string]::IsNullOrEmpty($Env:GH_TOKEN)) {
+                  Write-Error "Token file exists but is empty"
+                  exit 1
+              }
+              gh config set prompt disabled
+              $Env:GH_TOKEN | gh auth login --with-token
+              gh auth status
+            } else {
+                Write-Error "Token file not found at $HOME/.gh-token"
+                exit 1
+            }
+          - |
+            if ($Env:SEMAPHORE_ORGANIZATION_URL -like "*.semaphoreci.com") {
+              Write-Output "Skipping Vault setup for Semaphore CI"
+            } else {
+              $Env:VAULT_ADDR = "https://vault.cireops.gcp.internal.confluent.cloud"
+              $token = $(vault write -field=token "auth/semaphore_self_hosted/login" role="default" jwt="$Env:SEMAPHORE_OIDC_TOKEN")
+              vault login -no-print token=$token
+            }
+          - $E2E_USERNAME = $(vault kv get -field=E2E_USERNAME v1/ci/kv/vscodeextension/testing)
+          - Set-Content -Path .env -Value "E2E_USERNAME=$E2E_USERNAME"
+          - $E2E_PASSWORD = $(vault kv get -field=E2E_PASSWORD v1/ci/kv/vscodeextension/testing)
+          - Add-Content -Path .env -Value "E2E_PASSWORD=$E2E_PASSWORD"
       jobs:
+        - name: "Build & Test (VS Code)"
+          commands:
+            - npm ci --prefer-offline --include=dev
+            - npx playwright install
+            - npx gulp build
```

Comment:
We can't run [`gulp ci`](https://github.com/confluentinc/vscode/blob/8a87fec6bd710ab1e6e0a6e1cfc8df4016cb5035/Gulpfile.js#L34) here because `lint` will fail until https://github.com/confluentinc/vscode/pull/1246 is merged, and `check` will throw ~11 TS errors (even though we get zero TS errors on non-Windows agents).